src/vos/vos_io.c

```diff
@@ -1971,6 +1972,25 @@ vos_reserve_scm(struct vos_container *cont, struct vos_rsrvd_scm *rsrvd_scm,
 		umoff = umem_alloc(vos_cont2umm(cont), size);
 	}
 
+	vpm = cont->vc_pool->vp_metrics;
+	if (vpm && vpm->vp_space_metrics.vsm_scm_used) {
+		int		 rc;
+		struct vos_pool	*pool = vos_cont2pool(cont);
+		daos_size_t	 scm_used;
+
+		rc = pmemobj_ctl_get(pool->vp_umm.umm_pool,
+				     "stats.heap.run_allocated", &scm_used);
```

Comment:
I tried to use this metric (see 64a709367bfe9ae62c367d575aa87ae18e480815). It was working well with SCM-RAM but gives me some crazy number when I run it on CI with SCM-PMEM. It seems to be an spdk issue.

Reply:
You mean PMDK issue? We observed such issue before, and it's supposed to be fixed in the current PMDK version used by master. Could you file a ticket?

'run_allocated' is used for fragmentation estimation along with 'run_active', it doesn't include all allocations (see the PMDK man page).

Reply:
You say a crazy number with the latest pmdk ?   The issue is supposed to be fixed in 1.12.1.

Reply:
As the PR is not merged, I should have the trace of the numbers that I got.
For the end of this week it will be hard for me to work on this PR.
For next week. I will check these numbers, the version of pmdk, etc. 
I will complete this thread with the results of these investigations.

Reply:
For the crazy numbers: I was pushing 128 MiB of data with ior and I was seeing a space usage of 161GiB.
- The logs could be seen at https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-11572/6/testReport/junit/FTEST_telemetry/TelemetryPoolSpaceMetrics/1___telemetry_telemetry_pool_space_metrics_py_TelemetryPoolSpaceMetrics_test_telemetry_pool_space_metrics_run_container_hosts_ior_mpirun_pool_server_config_engines_0_storage_0_1_b473/
- This build was generated with the commit 65c63a9c216f819bbcd2b089c135891ca298db86 where SCM usage was also updated in `vos_space_query()` function with the call to pmemobj_ctl_get() function with the `"stats.heap.curr_allocated"` as parameter. This code was working well when SCM-RAM was used on my local WOLF machines.

Reply:
> You say a crazy number with the latest pmdk ? The issue is supposed to be fixed in 1.12.1.

@jolivier23 and @NiuYawei, I confirm that the PMDK issue is not fixed. More details on this issue could be found in the ticket https://daosio.atlassian.net/browse/DAOS-12968 that I have created to track this issue.